### DISCUSSION TITLE
Is there any more information about reach except official docs ?

### DISCUSSION THREAD

**Discussion by Confucian-e** [reach-sh/reach-lang#511] [Q&A] [answered]
I get some trouble in learning 'View' and 'Event', I wonder what they are used for. So where can I get more material except official docs?

**Comment by jeapostrophe** [comment, marked as answer]
The main source of information is the docs.

You use `View`s when you want agents external to the program (non-participants) to learn about the current values of the program. For example, an NFT program would expose the current owner as a `View`.

You use `Event`s when you want agents external to the program (non-participants) to learn about the history of the program. For example, an NFT program could emit an event every time the owner changed; thus, an external agent could see the history of ownership.

**Reply by Confucian-e** [reply]
Thanks
